db/compaction/compaction_picker_level.cc

```diff
@@ -448,6 +448,10 @@ bool LevelCompactionBuilder::PickFileToCompact() {
     // do not pick a file to compact if it is being compacted
     // from n-1 level.
     if (f->being_compacted) {
+      if (ioptions_.compaction_pri == kRoundRobin) {
+        // Advance the cursor if a file is being compacted
+        vstorage_->UpdateCompactCursor(start_level_, cmp_idx + 1);
```

Comment:
What if a file is skipped due to `being_compacted` in round-robin policy? Let's say file 1 is in the first position after sorting by round-robin. But since file 1 is firstly selected to be compacted together with some upper-level files. Then file 1 will be skipped and the next file (say file2) that is not being compacted will be selected. In that case, even though file 2 is compacted due to round-robin, we shall not advance the cursor since there are still some old data in file 1.

Reply:
Yes you are smarter than me as I only figured that out after trying my own suggestion, after which I added this comment: https://github.com/facebook/rocksdb/pull/10107#discussion_r895382982. By not skipping ahead to file 2 for the initial implementation we can avoid that problem.